prov/verbs/src/verbs_cm.c

```diff
@@ -276,12 +278,15 @@ fi_ibv_msg_ep_reject(struct fid_pep *pep, fid_t handle,
 	cm_hdr = alloca(sizeof(*cm_hdr) + paramlen);
 	fi_ibv_msg_ep_prepare_cm_data(param, paramlen, cm_hdr);
 
-	if (connreq->is_xrc)
+	if (connreq->is_xrc) {
+		fastlock_acquire(&_pep->eq->lock);
 		ret = fi_ibv_msg_xrc_ep_reject(connreq, cm_hdr,
 				(uint8_t)(sizeof(*cm_hdr) + paramlen));
-	else
+		fastlock_release(&_pep->eq->lock);
+	} else {
 		ret = rdma_reject(connreq->id, cm_hdr,
 			(uint8_t)(sizeof(*cm_hdr) + paramlen)) ? -errno : 0;
+	}
```

Comment:
Serialization is needed around rdma_reject as well.

Reply:
@a-ilango - Thanks, I updated the PR.